### PR TITLE
usm_ndarray.to_device(dev, stream=queue) support

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
This PR fixes two bugs: closes gh-1330, and fixes array_api_tests failure regarding `usm_ndarray.to_device` not supporting `stream` keyword.

- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, reproducer or referred to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: dpctl/tensor/_copy_utils.py [hunk 1]
```diff
@@ -279,7 +279,10 @@ def _copy_from_usm_ndarray_to_usm_ndarray(dst, src):
         common_shape = common_shape[ones_count:]
 
     if src.ndim < len(common_shape):
-        new_src_strides = (0,) * (len(common_shape) - src.ndim) + src.strides
+        pad_count = len(common_shape) - src.ndim
+        new_src_strides = (0,) * pad_count + tuple(
+            s if d > 1 else 0 for s, d in zip(src.strides, src.shape)
+        )
```
Comment (ndgrigorian) [inline dpctl/tensor/_copy_utils.py:285]: I wonder, is there a reason not to reuse the ``_broadcast_strides`` function from ``_manipulation_functions`` here?

I tested it out and (besides some importing issues) it seemed to give the same results, passed tests, etc.

Comment (oleksandr-pavlyk) [reply dpctl/tensor/_copy_utils.py]: I moved the implementation of `_broadcast_strides` to `_copy_utils` and imported it from `_manupulation_functions.py`.

**Comment by antonwolfy** [comment]
The following example is still not working:
```python
In [1]: import dpnp, numpy, dpctl, dpctl.tensor as dpt

In [2]: a = dpt.ones((1, 2, 1, 4))

In [3]: b = dpt.empty((2, 3, 4))

In [4]: b[...] = a
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[4], line 1
----> 1 b[...] = a

File dpctl/tensor/_usmarray.pyx:1178, in dpctl.tensor._usmarray.usm_ndarray.__setitem__()

File ~/miniconda3/envs/dpnp_dev/lib/python3.9/site-packages/dpctl/tensor/_copy_utils.py:307, in _copy_from_usm_ndarray_to_usm_ndarray(dst, src)
    305 else:
    306     src_same_shape = src
--> 307     src_same_shape.shape = common_shape
    309 _copy_same_shape(dst, src_same_shape)

File dpctl/tensor/_usmarray.pyx:576, in dpctl.tensor._usmarray.usm_ndarray.shape.__set__()

TypeError: Can not reshape array of size 8 into (2, 3, 4)

In [5]: dpctl.__version__
Out[5]: '0.14.6dev1+35.g5379d9342'

In [6]: conda list dpctl
# packages in environment at /home/xantvol/miniconda3/envs/dpnp_dev:
#
# Name                    Version                   Build  Channel
dpctl                     0.14.6dev1      py39h7bf5fec_35    file:///mnt/c/Users/antonvol/Downloads/dpctl Linux Python 3.9
```

**Comment by antonwolfy** [comment]
dpnp tests works fine with the PR, thank you @oleksandr-pavlyk !
I've filed a separate #1334 for remaining issues.

**Review by antonwolfy — APPROVED** [review]

**Review by ndgrigorian — APPROVED** [review]
I've looked through the PR and don't see any glaring issues.

LGTM, thanks @oleksandr-pavlyk !

**Pull request merged**